ctapipe/calib/camera/flatfield.py

```diff
@@ -165,28 +166,27 @@ def __init__(self, **kwargs):
         self.arrival_times = None  # arrival time per event in sample
         self.sample_masked_pixels = None  # masked pixels per event in sample
 
-    def _extract_charge(self, event):
+    def _extract_charge(self, event) -> DL1CameraContainer:
         """
         Extract the charge and the time from a calibration event
 
         Parameters
         ----------
         event : general event container
 
+        Returns
+        -------
+        DL1CameraContainer
         """
 
         waveforms = event.r1.tel[self.tel_id].waveform
         selected_gain_channel = event.r1.tel[self.tel_id].selected_gain_channel
 
         # Extract charge and time
-        charge = 0
-        peak_pos = 0
         if self.extractor:
-            charge, peak_pos = self.extractor(
-                waveforms, self.tel_id, selected_gain_channel
-            )
-
-        return charge, peak_pos
+            return self.extractor(waveforms, self.tel_id, selected_gain_channel)
+        else:
+            return DL1CameraContainer(image=0, peak_pos=0, is_valid=False)
```

Comment:
Shouldn't this be `None` or `np.zeros(n_pixels)`? Filling a single 0 seems wrong

Reply:
this comes from Franca's code, so I didn't want to change anything. I guess these values are never relied upon (later the is_valid flag is checked, so these are basically null events),  However, it's true the DataWriter would choke on them, so maybe better to do it right

Reply:
or maybe `image=None, peak_pos=None`